### PR TITLE
[11.0][MIG] web_widget_timepicker

### PR DESCRIPTION
**Pull request by Tardo**
Changes from 10.0:
- Removed Template
- Removed CSS
- Removed external libs, now uses date picker widget (bootstrap datetime-picker)
- Now it's based on 'FieldDate' instead of 'AbstractField'

### CODE REVIEW

File: web_widget_timepicker/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# -*- coding: utf-8 -*-
```
Comment (simahawk) [inline web_widget_timepicker/__init__.py]: remove coding declarations

File: web_widget_timepicker/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# © 2016 Vividlab (<http://www.vividlab.de>)
```
Comment (simahawk) [inline web_widget_timepicker/__manifest__.py]: (c) -> Copyright

File: web_widget_timepicker/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# © 2019 Alexandre Díaz (dev@redneboa.es)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Web Timepicker Widget",
+    "version": "11.0.1.0.0",
+    "author": "VividLab, "
+              "Odoo Community Association (OCA), "
```
Comment (simahawk) [inline web_widget_timepicker/__manifest__.py]: OCA should stay in the last place

File: web_widget_timepicker/readme/CREDITS.rst [hunk 4]
```diff
@@ -0,0 +1,3 @@
+* The module uses the `jquery-timepicker <https://github.com//jonthornton//jquery-timepicker#timepicker-plugin-for-jquery>`_. plugin by Jon Thornton. This software is made available under the open source MIT License. © 2014 Jon Thornton and contributors
```
Comment (simahawk) [inline web_widget_timepicker/readme/CREDITS.rst]: do we really need this? The std timepicker already uses bootstrap-timepicker https://github.com/odoo/odoo/blob/b6f11ace836e946eeeae110cdb08d16d80546238/addons/web/static/src/js/widgets/date_picker.js#L257

**Review by simahawk — CHANGES_REQUESTED** [review]

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+odoo.define('web.widget_timepicker', function (require) {
+    'use strict';
+
+    var field_registry = require('web.field_registry');
+    var field_utils = require('web.field_utils');
+    var basic_fields = require('web.basic_fields');
+    var FieldFloatTime = basic_fields.FieldFloatTime;
+
+    var FieldTimePicker = FieldFloatTime.extend({
+        cssLibs: ['web_widget_timepicker/static/src/lib/jquery-timepicker/jquery.timepicker.css'],
```
Comment (tarteo) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: Needs to be prefixed with '/' to make sure it loads correctly everywhere.

File: web_widget_timepicker/__manifest__.py (hunk 2 above)
Comment (Tardo) [reply web_widget_timepicker/__manifest__.py]: I'm not Vividlab... idk if I can change this... :S

File: web_widget_timepicker/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# © 2016 Vividlab (<http://www.vividlab.de>)
+# © 2019 Alexandre Díaz (dev@redneboa.es)
```
Comment (Tardo) [inline web_widget_timepicker/__manifest__.py]: But i go to change this

**Review by tarteo — CHANGES_REQUESTED** [review]

File: web_widget_timepicker/__manifest__.py (hunk 2 above)
Comment (tarteo) [reply web_widget_timepicker/__manifest__.py]: You can just replace © with 'Copyright'. I don't think it's a problem :smile:

**Comment by Tardo** [comment]
Ready to review :B

**Comment by nikul-serpentcs** [comment]
> Ready to review :B

@Tardo Squash Commits

**Comment by Tardo** [comment]
Really need do the squash? Can be done from github in one click :P
Reviewers can use "Files Changed" tab ;)

If is necessary i will do, no problem!

**Comment by simahawk** [comment]
you will have to, since you miss the whole history https://github.com/OCA/web/commits/10.0/web_widget_timepicker ;)
Follow the guide https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0

**Comment by Tardo** [comment]
Ok! i see.. i missed ```git format-patch --keep-subject --stdout origin/11.0..origin/10.0 -- $MODULE | git am -3 --keep```

I only need do ```git rebase -i HEAD~5```? This is where I'm going to ruin everything.
Sorry for the inconvenience of my noob questions :B

**Comment by simahawk** [comment]
```
# reset to 11.0
g reset --hard oca/11.0
# get the module from v10
g format-patch ....
# cherry pick your commit from reflog
g reflog |grep web_widget_timepicker
# find the commit hash
g cherry-pick $hash
```
You can even do it super-simple w/out cherry-pick (you might conflicts since you don't have atomic commits to cherry-pick):
```
# bkp your mig
cp -r  web_widget_timepicker web_widget_timepicker_v11
# reset to 11.0
g reset --hard oca/11.0
# get the module from v10
g format-patch ....
# restore your bkp
mv web_widget_timepicker_v11 web_widget_timepicker
# check the diff, stage changes, commit again
```

**Comment by Tardo** [comment]
@simahawk thx!! I hope I did it right xD

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 7]
```diff
@@ -1,89 +1,90 @@
-odoo.define('web_widget_timepicker', function (require) {
-    "use strict";
+odoo.define('web.widget_timepicker', function (require) {
```
Comment (simahawk) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: this is wrong, this JS module belongs to `web_widget_timepicker` and not to `web`

**Comment by Tardo** [comment]
@simahawk done, thx!

**Comment by Tardo** [comment]
Launching flare... What does this PR need to close?

**Comment by simahawk** [comment]
> Launching flare... What does this PR need to close?

only time to get back to them all... :man_shrugging:

File: web_widget_timepicker/static/src/js/web_widget_timepicker.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+            }
+        },
+
+        setValue: function (value) {
+            this.set({'value': value});
+            var formatted_value = value ? this._formatClient(value) : null;
+            this.$input.val(formatted_value);
+            if (this.picker) {
+                var fdate = new moment(formatted_value, this.options.format);
+                this.picker.date(fdate&&fdate.isValid()?fdate:new moment());
```
Comment (simahawk) [inline web_widget_timepicker/static/src/js/web_widget_timepicker.js]: can we put some space between these operators?

**Review by simahawk — COMMENTED** [review]
LG (code review only).
Only nitpicking needed (see my comment).
Finaly, pls squash your commits altogether and amend the msg as `[11.0][MIG] web_widget_timepicker`

**Comment by Tardo** [comment]
@simahawk done ;)

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
@tarteo fine for you?

**Comment by tarteo** [comment]
Yes :+1:

**Review by tarteo — APPROVED** [review]

**Pull request merged**